pubsub-mapped-api/src/main/java/com/google/pubsub/clients/config/ProducerConfig.java

```diff
@@ -0,0 +1,227 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+package com.google.pubsub.clients.config;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.Map;
+import java.util.Set;
+import java.util.HashMap;
+import java.util.Properties;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.AbstractConfig;
+import org.apache.kafka.common.config.ConfigDef.Type;
+import org.apache.kafka.common.config.ConfigDef.Range;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.common.config.ConfigDef.Importance;
+import org.apache.kafka.common.config.ConfigDef.ValidString;
+
+/**
+ * Provides the configurations for a KafkaProducer instance.
+ */
+public class ProducerConfig extends AbstractConfig {
```

Comment:
I created a separate branch with the refactored configs, utilizing Kafka's configs as is, and CPS's as a separate layer over it.